sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -257,7 +257,7 @@ static void _send(gnrc_pktsnip_t *pkt)
 
     /* IP should not send anything here if it is not a 6LoWPAN interface,
      * so we don't need to check for NULL pointers */
-    if (datagram_size <= iface->max_frag_size) {
+    if (gnrc_pkt_len(pkt2->next) <= iface->max_frag_size) {
```

Comment:
And every subsequent usage of `datagram_size` is valid? So it is only this case where the "updated" size is necessary? Line 268 e.g. uses the unmodified size to compare it to `SIXLOWPAN_FRAG_MAX_LEN`.